tick/eval/node_evaluator.go

```diff
@@ -0,0 +1,97 @@
+package eval
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/influxdata/kapacitor/tick"
+)
+
+// ErrTypeGuardFailed is returned when a speicifc value type is requested thorugh NodeEvaluator (for example: "Float64Value")
+// when the node doesn't support the given type, for example "Float64Value" is called on BoolNode
+type ErrTypeGuardFailed struct {
+	RequestedType ValueType
+	ActualType    ValueType
+}
+
+func CreateTypeGuardFailedError(requested, actual ValueType) error {
```

Comment:
Also I would prefer to remove this function and use the composite syntax to create the error. That way you don't have to remember which arg is the requested and which is actual.

```go
ErrTypeGaurdFailed{RequestedType: r, ActualType: a}
```